lib/active_storage_support/base64_attach.rb

```diff
@@ -10,9 +10,12 @@ def attachment_from_data(attachment)
     end
 
     def fill_attachment_data(attachment, base64_data)
-      return unless base64_data.try(:is_a?, String) && base64_data =~ /^data:(.*?);(.*?),(.*)$/
+      return unless base64_data.try(:is_a?, String) && base64_data.strip.start_with?('data')
 
-      attachment[:io] = StringIO.new(Base64.decode64(Regexp.last_match(3)))
+      headers, data = base64_data.split(',')
+      headers =~ /^data:(.*?);(.*?)$/
```

Comment:
be careful, the `;` is also optional `data:[<mediatype>][;base64],<data>` https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs

Reply:
Changed!